### PR TITLE
Unobsolete GetAccountsAsync

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes #

**Changes proposed in this request**
Remove the obsolete attribute from the GetAccountsAsync method in the IConfidentialClientApplication interface to make it available for ROPC silent flow in Id web

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

**Review by trwalke — APPROVED** [review]

**Review by bgavrilMS — APPROVED** [review]

**Comment by bgavrilMS** [comment]
@jmprieur and @localden - can you pls provide a perspective on this? `GetAccounts` is still used safely in all other MSALs in both web app and ROPC scenarios.

File: src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs [hunk 1]
```diff
@@ -96,8 +96,6 @@ AcquireTokenByAuthorizationCodeParameterBuilder AcquireTokenByAuthorizationCode(
         /// <summary>
         /// Use <see cref="IClientApplicationBase.GetAccountAsync(string)"/> in web apps and web APIs, and use a token cache serializer for better security and performance. See https://aka.ms/msal-net-cca-token-cache-serialization.
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs:97]: Pls add a remark that this is only for use with ROPC

**Comment by localden** [comment]
@bgavrilMS @neha-bhargava - given that this is only for ROPC in CCA, no concerns on my part. This will effectively just list out the accounts in the cache?

File: src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs (hunk 1 above)
Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs:97]: we have no way to enforce that it's only used for ROPC in CCA, though
Are we sure that the cache is always partitioned?

**Comment by neha-bhargava** [comment]
> @bgavrilMS @neha-bhargava - given that this is only for ROPC in CCA, no concerns on my part. This will effectively just list out the accounts in the cache?

Yes, this will list the accounts in the cache. Only recommended for ROPC.

File: src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs (hunk 1 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs:98]: AFAIK the cache is always partitioned.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs:98]: It is partitioned. But please add a unit test (or update the integration test) @neha-bhargava, where you show that TokenCacheNotificationArgs has the correct `SuggestedCacheKey`.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs:98]: @neha-bhargava and I had a chat and concluded that `AcquireTokenForUsernamePassword` is partitioned, but `GetAccounts()` is not partitioned (and can't really be partitioned).

From MSAL perspective, we can ask ppl to use `GetAccount(id)`. 
For Id.Web, this gets a bit tricky.

Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs:98]: How would that work in MSAL? where do they get the id from? (assuming this is tid.oid?)?

Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs:98]: Proposing that CCA.AcquireTokenByUserPassword first checks the cache, and if not found or error, goes to ESTS.
Like OBO does.

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
We decided against it.

**Comment by neha-bhargava** [comment]
Not needed anymore.

**Pull request closed without merging**